dashboard/src/shared/Secret.ts

```diff
@@ -40,6 +40,39 @@ export default class Secret {
     return data;
   }
 
+  public static async createPullSecret(
+    name: string,
+    user: string,
+    password: string,
+    email: string,
+    server: string,
+    namespace: string,
+  ) {
+    const url = Secret.getLink(namespace);
+    const dockercfg = {
+      auths: {
+        [server]: {
+          username: user,
+          password,
+          email,
+          auth: btoa(`${user}:${password}`),
+        },
+      },
+    };
+    const { data } = await axiosWithAuth.post<ISecret>(url, {
+      apiVersion: "v1",
+      data: {
```

Comment:
You can also use the write-only `stringData` key and let k8s do the base64 encoding itself (a bit safer, given different encodings etc.)